src/audio-view.js

```diff
@@ -0,0 +1,193 @@
+
+*/
+
+const _ = require('lodash');
+
+const SculptureStore = require('@anyware/game-logic/lib/sculpture-store');
+const GAMES = require('@anyware/game-logic/lib/constants/games');
+import {Sound, VCFSound} from './audio-api';
+
+export default class AudioView {
+  constructor(store, config, dispatcher) {
+    this.store = store;
+    this.config = config;
+  }
+
+  // Loads all sounds, calls callback([err]) when done
+  load(callback) {
+    // Maps logical sound identifiers to filenames. We'll load these sounds next.
+    this.sounds = {
+      alone: {
+        ambient: new VCFSound({url: 'sounds/Alone_Mode/Pulse_Amb_Loop.wav', fadeIn: 3}),
+        handshake: 'sounds/Alone_Mode/Hand_Shake_01.wav'
+      },
+      mole: {
+        success: 'sounds/Game_01/G01_Success_01.wav',
+        failure: 'sounds/Game_01/G01_Negative_01.wav',
+        panels: [0,1,2].map(stripId => _.range(10).map(panelId => `sounds/Game_01/G01_LED_${("0"+(stripId*10+panelId+1)).slice(-2)}.wav`))
+      },
+//      disk: {
+//      },
+      simon: {
+        panels: [0,1,2].map(stripId => _.range(10).map(panelId => `sounds/Game_03/G03_LED_${("0"+(stripId*10+panelId+1)).slice(-2)}.wav`)),
+        success: 'sounds/Game_03/G03_Success_01.wav',
+        failure: 'sounds/Game_03/G03_Negative_01.wav',
+        show: 'sounds/Game_03/G03_Light_Show_01.wav'
+      }
+    };
+
+    // Traverse this.sounds and replace the filenames with valid sound objects.
+    this._promises = [];
+    this._traverse(this.sounds);
+
+    // _loadSound() will create promises. We call the callback once all promises resolve
+    console.log(`${this._promises.length} promises`);
+    Promise.all(this._promises)
+      // Don't listen to events until we've loaded all sounds
+      .then(() => this.store.on(SculptureStore.EVENT_CHANGE, this._handleChanges.bind(this)))
+      .then(() => callback(null))
+      .catch(callback.bind(null));
+  }
+
+  // Traverses sound config objects and replaces nodes with valid, loaded, sounds
+  _traverse(node) {
+    for (let key in node) {
+      const value = node[key];
+      let sound;
+      if (typeof value === 'string') sound = node[key] = new Sound({url: value});
+      else if (value instanceof Sound) sound = value;
+      if (sound) this._loadSound(sound);
+      else this._traverse(value);
+    }
+  }
+
+  // Returns a promise to provide a comple, usable sound.
+  _loadSound(sound) {
+    const promise = sound.load();
+    this._promises.push(promise);
+    return promise;
+  }
+
+  _handleChanges(changes) {
+    if (this._animating) {
+      return;
+    }
+
+    if (changes.currentGame === GAMES.HANDSHAKE) this.sounds.alone.ambient.play();
+
+    this._handleHandshakeChanges(changes);
+    this._handleStatusChanges(changes);
+    this._handleLightChanges(changes);
+  }
+
+  _handleHandshakeChanges(changes) {
+    if (changes.handshakes) {
+      // Did someone shake my hand?
+      if (changes.handshakes[this.config.username]) {
```

Comment:
This doesn't account for multiple handshakes being activated or deactivated. Also, simply testing the truthiness of this will fail when other user's handshake and you will have undesired sounds playing all the time. Please see handshake-view.js in sculpture-client for an example of handling `changes.handshakes`.